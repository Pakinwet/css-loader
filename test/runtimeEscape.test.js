/* eslint-env mocha*/

const runtimeEscape = require('../src/runtimeEscape');

describe('runtimeEscape', () => {
  it('escape', () => {
    expect(runtimeEscape(true)).toMatchSnapshot(true);
    expect(runtimeEscape('image.png')).toMatchSnapshot();
    expect(runtimeEscape('"image.png"')).toMatchSnapshot();
    expect(runtimeEscape("'image.png'")).toMatchSnapshot();
    expect(runtimeEscape('image other.png')).toMatchSnapshot();
    expect(runtimeEscape('"image other.png"')).toMatchSnapshot();
    expect(runtimeEscape("'image other.png'")).toMatchSnapshot();
    expect(runtimeEscape('image"other.png')).toMatchSnapshot();
    expect(runtimeEscape('image\nother.png')).toMatchSnapshot();
  });
});
