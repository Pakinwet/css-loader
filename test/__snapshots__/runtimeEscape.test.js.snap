// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`runtimeEscape escape 1`] = `true`;

exports[`runtimeEscape escape 2`] = `"image.png"`;

exports[`runtimeEscape escape 3`] = `"image.png"`;

exports[`runtimeEscape escape 4`] = `"image.png"`;

exports[`runtimeEscape escape 5`] = `"\\"image other.png\\""`;

exports[`runtimeEscape escape 6`] = `"\\"image other.png\\""`;

exports[`runtimeEscape escape 7`] = `"\\"image other.png\\""`;

exports[`runtimeEscape escape 8`] = `"\\"image\\\\\\"other.png\\""`;

exports[`runtimeEscape escape 9`] = `"\\"image\\\\nother.png\\""`;
